test/cppsuite/test_harness/workload/database_model.cxx

```diff
@@ -0,0 +1,143 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include <chrono>
+#include <tuple>
+
+#include "../util/api_const.h"
+#include "database_model.h"
+#include "random_generator.h"
+
+namespace test_harness {
+/* collection class implementation */
+collection::collection(const uint64_t id, const uint64_t key_count, const std::string &name)
+    : id(id), _key_count(key_count), name(name)
+{
+}
+
+uint64_t
+collection::get_key_count() const
+{
+    return (_key_count);
+}
+
+void
+collection::increase_key_count(uint64_t increment)
+{
+    _key_count += increment;
+}
+
+/* database class implementation */
+std::string
+database::build_collection_name(const uint64_t id)
+{
+    return (std::string("table:collection_" + std::to_string(id)));
+}
+
+void
+database::add_collection(uint64_t key_count)
+{
+    std::lock_guard<std::mutex> lg(_mtx);
+    if (_session.get() == nullptr)
+        _session = connection_manager::instance().create_session();
+    uint64_t next_id = _next_collection_id++;
+    std::string collection_name = build_collection_name(next_id);
+    /* FIX-ME-Test-Framework: This will get removed when we split the model up. */
```

Comment:
@luke-pearson Not related to this patch but didn't the model get split up? I'm wondering whether we can do something about this.

Reply:
Right yeah it was because the collection class had a mutex in it which couldn't be move copied, so we had to do this 🧠 fix.

In theory we can do the much simpler collection(next_id, key_count, collection_name). I'll give a try and then push it up if it works.